### ISSUE TITLE
twitch.tv Commercial break in progress with Twitch Turbo cookie

### ISSUE THREAD

**Issue by challi1337** [jrudess/streamdvr#160]
Hello,

I've been using this program for a while now, but it seems that when a streamer runs an ad manually I still get the ad even though my account which I give the program with the http cookie should not get ads because of my twitch turbo subscription.

Anyone know a fix or workaround for this?

**Comment by challi1337** [comment]
Yes like you've pointed out I'm using streamlink fetch and recorder.

I give the api_token and persistent cookies via '--http-cookie=.....' in the fetch and since in the past that helped already for recording (no prerolls) I didn't think of adding it also in the recorder. I use it for another website too (afreecatv) and works fine there aswell. But I'll give it a try with the argument added for both fetch and recorder. I'll update once I see results tomorrow

**Comment by challi1337** [comment]
Reopened since I found a fix for it.

So streamlink doesn't like just getting the cookie thats usually required to identify your account since they have a special argument just for twitch login which I found in their documentation. [link
](https://streamlink.github.io/latest/cli/plugins/twitch.html#authentication)
by adding the OAuth token it seems to have fixed the issue, not sure if necessary in fetch and recorder but I added it in both just to be save.
